### PR TITLE
Implemented endpoint to remove a rss feed from your channel

### PR DESCRIPTION
**Pull request by devos50**
This PR implements the endpoint to remove a rss feed from your channel. Also, I completely refactored the endpoint to add a RSS feed in the process to make it more REST.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis ready for review

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 1]
```diff
@@ -126,14 +126,11 @@ def render_GET(self, request):
 
         return json.dumps({"rssfeeds": feeds_list})
 
-    def render_PUT(self, request):
+    def modify_rss_feeds(self, request, add):
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: what are the params? Please update the doc

**Comment by lfdversluis** [comment]
comments givem

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py (hunk 1 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: I've added a short description in the doc under the method.

**Comment by devos50** [comment]
@whirm processed feedback, ready for your review!

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 2]
```diff
@@ -144,6 +142,37 @@ def render_PUT(self, request):
             request.setResponseCode(http.BAD_REQUEST)
             return json.dumps({"error": "rss_feed_url parameter missing"})
 
-        channel_obj.create_rss_feed(parameters['rss_feed_url'][0])
+        rss_feed_url = parameters['rss_feed_url'][0]
+
+        if add:
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Having a function that does completely opposite things depending on an argument is not really nice. Why don't you move the code in the if/else block to each `render_*` handler instead?

**Comment by devos50** [comment]
@whirm this day, we had a discussion about making this endpoint more RESTful and putting the rss feed parameter inside the URL. Is this really how we should do it? Can you give me an example of how the URL should look like when adding/removing a RSS feed?

**Comment by whirm** [comment]
Already resolved via IRC

**Comment by devos50** [comment]
@whirm implemented your feedback regarding the URLs. See the last three fixup commits.

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 3]
```diff
@@ -22,6 +22,14 @@ def return_404(request, message="your channel has not been created"):
         request.setResponseCode(http.NOT_FOUND)
         return message
 
+    def get_my_channel_object(self):
+        """
+        Returns the Channel object associated with you channel that is used to manage rss feeds.
+        """
+        my_channel_id = self.channel_db_handler.getMyChannelId()
+        channel_obj = self.session.lm.channel_manager.get_my_channel(my_channel_id)
+        return channel_obj
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: the last two lines should be merged

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 4]
```diff
@@ -126,53 +129,41 @@ def render_GET(self, request):
...
-
-        if add:
-            channel_obj.create_rss_feed(rss_feed_url)
-        else:
-            if rss_feed_url not in channel_obj.get_rss_feed_url_list():
-                request.setResponseCode(http.BAD_REQUEST)
-                return json.dumps({"error": "this url is not added to your RSS feeds"})
-            channel_obj.remove_rss_feed(rss_feed_url)
-
+        channel_obj.create_rss_feed(self.feed_url)
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Shouldn't we check if the feed is already there?

File: Tribler/Core/Modules/restapi/my_channel_endpoint.py [hunk 5]
```diff
@@ -126,53 +129,41 @@ def render_GET(self, request):
...
-        return self.modify_rss_feeds(request, add=True)
+        channel_obj = self.get_my_channel_object()
+        if channel_obj is None:
+            return MyChannelBaseEndpoint.return_404(request)
 
-    def render_DELETE(self, request):
-        """
-        Remove a RSS feed from your channel.
+        if self.feed_url not in channel_obj.get_rss_feed_url_list():
+            request.setResponseCode(http.BAD_REQUEST)
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/my_channel_endpoint.py]: Shouldn't this be a 404 you are trying to delete a non existing resource

**Comment by devos50** [comment]
@whirm I've implemented your feedback. Can you take a look at my last fixup commits?

File: doc/Tribler REST API.md [hunk 6]
```diff
@@ -135,11 +135,11 @@ Returns a list of rss feeds in your channel. Each rss feed items contains the UR
 
 ## `PUT /mychannel/rssfeed/{feedurl}`
 
-Add a RSS feed to your channel. Note that the rss feed url should be URL-encoded.
+Add a RSS feed to your channel. Returns error 409 if the supplied RSS feed already exists. Note that the rss feed url should be URL-encoded.
```
Comment (whirm) [inline doc/Tribler REST API.md]: I would add the status name too so people don't have to look up for them.

**Comment by whirm** [comment]
Looks great, squash!

**Comment by devos50** [comment]
```
test_channel_search (Tribler.Test.test_remote_search.TestRemoteChannelSearch) ... Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/RemoteTorrentHandler.py", line 472, in _success_callback
    tdef = TorrentDef.load_from_dict(meta_info)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/TorrentDef.py", line 153, in load_from_dict
    return TorrentDef._create(metainfo)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/TorrentDef.py", line 109, in _create
    maketorrent.copy_metainfo_to_input(t.metainfo, t.input)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/APIImplementation/maketorrent.py", line 334, in copy_metainfo_to_input
    outpath = pathlist2filename(file['path'])
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/APIImplementation/maketorrent.py", line 246, in pathlist2filename
    return fullpath.decode('utf-8')
  File "/usr/lib/python2.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
exceptions.UnicodeDecodeError: 'utf8' codec can't decode byte 0xbd in position 92: invalid start byte
```

Appears to be an error in the torrent def a peer received from another peer...

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
```
```

**Comment by devos50** [comment]
retest this please

**Comment by whirm** [comment]
> test_confirmationdialog (Tribler.Test.test_gui_dialogs.TestGuiDialogs) ... watchog wx thread *******TRIPPED!******, hasn't been set for 10.0495 secs.

what happened there?

**Comment by devos50** [comment]
@whirm see #2204, I'm not sure what happened there

**Comment by devos50** [comment]
Last run failed due to tunnel tests timing out...

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
I aborted the previous build

**Comment by devos50** [comment]
@whirm this PR is ready.

**Pull request merged**